### PR TITLE
ability to parse chrome eval exceptions

### PR DESCRIPTION
**Pull request by sinegovsky-ivan**
Hello!
At now Library has a problems with parsing V8 eval stack frames.
For example: 
```
ERROR Error: test
    at eval (eval at get isAppReady (app_store.ts:24), <anonymous>:1:7)
    at AppStore.get isAppReady (app_store.ts:24)
    at trackDerivedFunction (mobx.module.js:761)
    at ComputedValue.push../node_modules/mobx/lib/mobx.module.js.ComputedValue.computeValue (mobx.module.js:1258)
```
will be unparsed.

this PR fixes this behavior.

### CODE REVIEW

File: lib/src/frame.dart [hunk 1]
```diff
@@ -22,6 +22,8 @@ final _v8Frame =
 
 // http://pub.dartlang.org/stuff.dart.js:560:28
 final _v8UrlLocation = new RegExp(r'^(.*):(\d+):(\d+)|native$');
+// eval at Foo (main.dart.js:588)
+final _v8UrlEvalLocation = new RegExp(r'^(.*):(\d+)');
```
Comment (sigmundch) [inline lib/src/frame.dart]: I'd prefer if we make the eval pattern below more accurate. That is, update _v8EvalLocation below to handle an optional column instead. As a result, the new logic would only apply on lines that contain `eval at`.

File: test/frame_test.dart [hunk 2]
```diff
@@ -228,8 +228,6 @@ void main() {
           'Foo (dart:async/future.dart:10:15)');
       expectIsUnparsed(
           (text) => new Frame.parseV8(text), '    at dart:async/future.dart');
-      expectIsUnparsed((text) => new Frame.parseV8(text),
```
Comment (sigmundch) [inline test/frame_test.dart]: Let's bring back this test once we address the other comment.

File: lib/src/frame.dart (hunk 1 above)
Comment (sinegovsky-ivan) [reply lib/src/frame.dart]: @sigmundch i've updated. But regexp has no matches.
This is strange, because JS version of regexp handles matches: https://regex101.com/r/GLygY7/2

any issues with my regexp?

Comment (sigmundch) [reply lib/src/frame.dart]: I can't see the updated code in Dart. Is the commit uploaded?

Comment (sinegovsky-ivan) [reply lib/src/frame.dart]: @sigmundch i've found solution. please check updated regexp

File: test/frame_test.dart [hunk 3]
```diff
@@ -258,12 +254,13 @@ baz@http://pub.dartlang.org/buz.js:56355:55
       expect(trace.frames[2].member, equals('baz'));
     });
 
-    test('parses a Firefox stack trace with nested evals in anonymous function',
+    test(
+        'parses a Firefox stack trace with nested evals in anonymous function correctly',
```
Comment (sigmundch) [inline test/frame_test.dart]: undo these unrelated changes?

File: test/frame_test.dart [hunk 4]
```diff
@@ -223,12 +223,8 @@ void main() {
           (text) => Frame.parseV8(text), '    at Foo (dart:async/future.dart)');
       expectIsUnparsed((text) => Frame.parseV8(text),
           '    at (dart:async/future.dart:10:15)');
-      expectIsUnparsed(
```
Comment (sigmundch) [inline test/frame_test.dart]: can this test case remain? I'd expect we'd continue to leave this unparsed.

File: lib/src/frame.dart [hunk 5]
```diff
@@ -21,7 +21,7 @@ final _v8Frame =
     RegExp(r'^\s*at (?:(\S.*?)(?: \[as [^\]]+\])? \((.*)\)|(.*))$');
 
 // http://pub.dartlang.org/stuff.dart.js:560:28
-final _v8UrlLocation = RegExp(r'^(.*):(\d+):(\d+)|native$');
+final _v8UrlLocation = RegExp(r'^(.*?):(\d+)(?::(\d+))?|native$');
```
Comment (sigmundch) [inline lib/src/frame.dart:25]: I apologize, when I made the earlier comment I didn't realize that the column/line was extracted here and not in `_v8EvalLocation`, now I see why you had the separate regexp initially. I was hoping that we could ignore the column only in the context of an eval, but I see now that it is much trickier with the code as it is written below.

The current solution looks good. Just a couple questions:
* can we omit the extra `?` on the first group?
* if it is the same below, we could also write this with an additional `|` to list the three cases separately:
  `^(.*):(\d+):(\d+)|native|(.*):(\d+)$`. If you find that more readable, feel free to switch.

File: lib/src/frame.dart [hunk 6]
```diff
@@ -21,7 +21,7 @@ final _v8Frame =
     RegExp(r'^\s*at (?:(\S.*?)(?: \[as [^\]]+\])? \((.*)\)|(.*))$');
 
 // http://pub.dartlang.org/stuff.dart.js:560:28
```
Comment (sigmundch) [inline lib/src/frame.dart]: let's add a second example here with no column

File: test/frame_test.dart [hunk 7]
```diff
@@ -258,12 +254,13 @@ baz@http://pub.dartlang.org/buz.js:56355:55
...
 
-    test('parses a Firefox stack trace with nested evals in anonymous function',
+    test(
+        'parses a Firefox stack trace with nested evals in anonymous function correctly',
         () {
       var trace = Trace.parse('''
         Foo._bar@http://pub.dartlang.org/stuff.js:18056:12
         anonymous@file:///C:/example.html line 7 > eval line 1 > eval:1:1
-        anonymous@file:///C:/example.html line 45 > Function:1:1
+        anonymous@file:///C:/example.html line 45 > Function:1:1
```
Comment (sigmundch) [inline test/frame_test.dart]: undo?

File: lib/src/frame.dart (hunk 5 above)
Comment (sinegovsky-ivan) [reply lib/src/frame.dart:25]: @sigmundch 
This won't work without extra `?` because of greedy match.

File: test/frame_test.dart (hunk 4 above)
Comment (sinegovsky-ivan) [reply test/frame_test.dart]: returned

File: lib/src/frame.dart (hunk 6 above)
Comment (sinegovsky-ivan) [reply lib/src/frame.dart]: added

File: pubspec.yaml [hunk 8]
```diff
@@ -1,5 +1,13 @@
 name: stack_trace
-version: 1.9.4-dev
+# Note! This version number is referenced directly in the pub source code in
```
Comment (sigmundch) [inline pubspec.yaml]: seems this comment was deleted on master before, you may need to rebase to avoid introducing this again?

File: lib/src/frame.dart [hunk 9]
```diff
@@ -237,6 +240,10 @@ class Frame {
           return Frame._parseFirefoxEval(frame);
         }
 
+        if (match[3].contains(' line ')) {
```
Comment (sigmundch) [inline lib/src/frame.dart]: please delete. seems like this was added accidentally?

**Review by sigmundch — COMMENTED** [review]
Thanks. LGTM, just minor cleanup below.

File: lib/src/frame.dart (hunk 9 above)
Comment (sinegovsky-ivan) [reply lib/src/frame.dart]: @sigmundch, yes, sorry. removed

File: pubspec.yaml (hunk 8 above)
Comment (sinegovsky-ivan) [reply pubspec.yaml]: did it.

**Review by sigmundch — APPROVED** [review]
Thanks!

**Comment by sinegovsky-ivan** [comment]
Thank you!

How long does it usually take between merge to master and publish to pub.dev?

**Comment by sigmundch** [comment]
It's case by case - @natebosch any concerns on publishing this relatively soon?

**Comment by natebosch** [comment]
> any concerns on publishing this relatively soon?

This looks like it broke some tests. I'll have to dig a bit.

**Comment by sigmundch** [comment]
Thanks @natebosch - I only validated that internal tests of the package were all passing before merging, but I didn't check downstream packages.

**Comment by natebosch** [comment]
Will be fixed by #69
